### PR TITLE
Require developers to explicitly request using a different Python 3 v…

### LINKED ISSUE #22724: Require developers to explicit request using different Python 3 version
I see many folks ending up using Python 3.8 and 3.9 by mistake in their venvs that causes Python package installation issues.

I will be changing it to require setting an env variable explicitly (`ALLOW_NON_DEFAULT_PYTHON_VERSION` or `PYTHON_VERSION`).

### PR DESCRIPTION
**Pull request by armenzg**
…ersion

The developer is now warned if they are using a Python version different than 3.6.
The developer will need to explicitely set `PYTHON_VERSION` if they wish to use a different
Python version.

Fixes #22724

### CODE REVIEW

File: scripts/ensure-venv.sh [hunk 1]
```diff
@@ -40,10 +41,26 @@ EOF
...
-        # At the moment, Pillow does not install with Python 3.9 and requires adding more build dependencies.
-        [[ "$minor" -ge 9 ]] &&
-            die "We recommend you create this virtualenv with a Python version between 3.6 and 3.8. Remove $VIRTUAL_ENV and start again."
+        # If .venv is created with Python greater than 3.6 you might encounter problems and we want to ask you to downgrade
+        # unless you explicitely set an environment variable
+        if [[ "$minor" -gt 6 ]]; then
+            if [[ -n "$PYTHON_VERSION" ]]; then
+                cat << EOF
+${yellow}${bold}
+You have explicitely set a non-recommend Python version (${PYTHON_VERSION}). You're on your own.
```
Comment (joshuarli) [inline scripts/ensure-venv.sh:50]:
```suggestion
You have explicitly set a non-recommended Python version (${PYTHON_VERSION}). You're on your own.
```

**Comment by joshuarli** [comment]
I feel like `SENTRY_PYTHON_VERSION` would be better, `PYTHON_VERSION` is generic and it's a good practice to namespace environment variables so we don't somehow affect other tooling.

**Comment by armenzg** [comment]
Thank you, Josh! Good suggestion.

File: scripts/ensure-venv.sh [hunk 2]
```diff
@@ -40,10 +41,26 @@ EOF
...
+${yellow}${bold}
+You have explicitly set a non-recommended Python version (${SENTRY_PYTHON_VERSION}). You're on your own.
+${reset}
+EOF
+            else
+                cat << EOF
+${red}${bold}
+WARNING! You are running a virtualenv with a Python version different than 3.6
+We recommend you start with a fresh virtualenv or to set the variable SENTRY_PYTHON_VERSION
+to the Python version you want to use (e.g. 3.7).
```
Comment (joshuarli) [inline scripts/ensure-venv.sh:58]:
```suggestion
ERROR! You are running a virtualenv with a Python version different than 3.6
We recommend you start with a fresh virtualenv or to set the variable SENTRY_PYTHON_VERSION
to the Python version you want to use (e.g. 3.7).
```

**Review by joshuarli — APPROVED** [review]
I tried this out, it works.

**Pull request merged**